### PR TITLE
docs: update the roadmap to reflect the progress as of v17

### PR DESCRIPTION
**Pull request by mgechev**

### CODE REVIEW

File: aio/content/guide/roadmap.md [hunk 1]
```diff
@@ -14,37 +14,36 @@ For example, we release features in the next minor after completion or the next
...
+We are working on making it possible to debug and profile Angular apps embedded within an iframe on the page. This feature will allow you to select an iframe and inspect it directly within Angular DevTools.
 
-### Introduce dependency injection debugging APIs
+### Enabling hybrid rendering by default
 
-To improve the debugging utilities of Angular and Angular DevTools, we'll work on APIs that provide access to the dependency injection runtime. As part of the project, we'll expose debugging methods that allow us to explore the injector hierarchy and the dependencies across their associated providers. As of v16, we have a design of a feature that enables us to plug into the dependency injection life-cycle. As the next step, we'll implement the functionality and provide integration with Angular DevTools.
+We are working on more developer experience improvements which will allow us to enable hybrid rendering by default for new projects. In particular, we're focused on route-level rendering strategy configuration and improving developer experience for i18n support.
```
Comment (AndrewKushnir) [inline aio/content/guide/roadmap.md:38]: nit: I think it'd be great to give a quick definition (1 sentence) of what "hybrid rendering" means.

**Review by AndrewKushnir — APPROVED** [review]
LGTM, just a couple comments.

File: aio/content/guide/roadmap.md [hunk 2]
```diff
@@ -14,37 +14,36 @@ For example, we release features in the next minor after completion or the next
...
+This project rethinks the Angular reactivity model by introducing Signals as a reactivity primitive.
+Fully implemented the project will make Zone.js optional. The initial planning resulted in hundreds of discussions, conversations with developers, feedback sessions, user experience studies, and a series of [RFCs](https://github.com/angular/angular/discussions/49685), which received over 1,000 comments.
 
-As the next step, we will iterate on polishing full hydration and further explore the dynamically evolving space of partial hydration and resumability. These more advanced patterns carry their own trade-offs; we'll share updates as we progress.
+As part of the v17 release, we graduated the Angular Signals library and RxJS interoperability from developer preview. Next we’ll continue implementing the proposals from the RFC. The first steps will be introducing signal-based inputs and queries.
 
-### Improve runtime performance and developer experience with a new reactivity model
+### Explore ergonomic component-level code-splitting APIs
```
[the remark this answers is not in the document]
Comment (jelbourn) [reply aio/content/guide/roadmap.md:24]: I might also drop "Explore" since we're beyond that phase now

**Review by jelbourn — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]

File: aio/content/guide/roadmap.md [hunk 3]
```diff
@@ -56,22 +55,36 @@ We are evaluating components in Angular Material against accessibility standards
...
+To provide better customization of our Angular material components and enable Material 3 capabilities, we'll be collaborating with Google's Material Design team on defining token-based theming APIs. As of Q4 2023, we're refactoring components to use the new API, finalizing the comprehensive set of tokens, and updating the Sass API based on the new tokens.
 
 ### Modernize Angular's unit testing tooling
 
 In v12, we revisited the Angular end-to-end testing experience by replacing Protractor with modern alternatives such as Cypress, Nightwatch, and Webdriver.io. Next, we'd like to tackle `ng test` to modernize Angular's unit testing experience. In Q2, we introduced experimental [Jest](https://jestjs.io/) support and [announced](https://blog.angular.io/moving-angular-cli-to-jest-and-web-test-runner-ef85ef69ceca) the transition from Karma to the [Web Test Runner](https://modern-web.dev/docs/test-runner/overview/).
 
 ## Future
```
Comment (dgp1130) [inline aio/content/guide/roadmap.md:68]: Do we want to add something about route visualization in DevTools?

File: aio/content/guide/roadmap.md [hunk 4]
```diff
@@ -56,22 +55,36 @@ We are evaluating components in Angular Material against accessibility standards
 
 ### Investigate micro frontend architecture for scalable development processes
```
Comment (dgp1130) [inline aio/content/guide/roadmap.md:56]: Is this something we want to call "Done" for now? The scope was limited to "Investigate" which we have mostly done. I think getting MFE functionality in `application` builder comparable to the existing `browser` builder is a separate problem which is in-scope for `application` builder itself.

Comment (mgechev) [reply aio/content/guide/roadmap.md:56]: I'm still hoping to get time in December for a blog post.

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit dea16201daad11e7a10ed2ec284d28e65011f25e.

**Pull request closed without merging**